lib/cosmos/api/tlm_api.rb

```diff
@@ -571,11 +548,13 @@ def tlm_process_args(args, function_name, scope: $cosmos_scope, token: $cosmos_t
         latest = 0
         Store.instance.get_telemetry(target_name).each do |packet|
           item = packet['items'].find { |item| item['name'] == item_name }
-          # TODO Sort through items and return latest
-          # if item
-          #   time = Store.instance.get_tlm_item(target_name, packet['name'], 'PACKET_TIMESECONDS', scope: scope)
-          #   time = packet['items'].find { |item| item['name'] == 'PACKET_TIMESECONDS' }
-          # end
+          if item
+            _, msg_hash = Store.instance.get_oldest_message("#{scope}__DECOM__#{target_name}__#{packet['packet_name']}")
+            if msg_hash && msg_hash['time'] && msg_hash['time'].to_i > latest
+              packet_name = packet['packet_name']
+              latest = msg_hash['time'].to_i
+            end
+          end
```

Comment:
We talked about handling LATEST and maybe I can use get_oldest_message or read_topic_last here directly rather than the initial call to Store.instance.get_telemetry